### PR TITLE
Add persistence to Mock Service Worker implementation

### PR DESCRIPTION
**Pull request by praseodym**
Supports data entry resume, abort, and status.

Also add a new 'not available in mock' page that used to hide the finalisation page, because PDF creation doesn't work in the mock version.

### CODE REVIEW

**Review by jorisleker — APPROVED** [review]

File: frontend/app/module/NotAvailableInMock.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { Link } from "react-router-dom";
+
+import { NavBar } from "app/component/navbar/NavBar";
+
+export function NotAvailableInMock() {
```
Comment (lkleuver) [inline frontend/app/module/NotAvailableInMock.tsx:5]: maybe add a short description and/or title prop?

File: frontend/app/routes.tsx [hunk 2]
```diff
@@ -44,7 +45,7 @@ export const routes = createRoutesFromElements(
           <Route path="list/:listNumber" element={<CandidatesVotesPage />} />
           <Route path="save" element={<CheckAndSaveForm />} />
         </Route>
-        <Route path="finalise" element={<FinaliseElectionPage />} />
+        <Route path="finalise" element={__API_MSW__ ? <NotAvailableInMock /> : <FinaliseElectionPage />} />
```
Comment (lkleuver) [inline frontend/app/routes.tsx:48]:
```
<NotAvailableInMock title="Input Finalise" />
```

**Review by lkleuver — DISMISSED** [review]

**Review by cikzh — CHANGES_REQUESTED** [review]
Looks good! Only two minor things and a question

File: frontend/lib/api-mocks/PollingStationMockData.ts [hunk 3]
```diff
@@ -15,10 +15,11 @@ export const pollingStationMockData: PollingStation = {
 };
 
 export const getPollingStationMockData = (election_id: number): PollingStation[] => {
+  const offset = (election_id - 1) * 10;
```
[the remark this answers is not in the document]
Comment (praseodym) [reply frontend/lib/api-mocks/PollingStationMockData.ts:18]: Yes!

File: frontend/app/module/data_entry/page/FinaliseElectionPage.tsx [hunk 4]
```diff
@@ -58,7 +58,7 @@ export function FinaliseElectionPage() {
       <PageTitle title="Invoerfase afronden - Abacus" />
       <NavBar>
         <Link to={"/overview"}>Overzicht</Link>
-        <IconChevronRight />
+        <IconChevronRight style={{ fill: "white" }} />
```
[the remark this answers is not in the document]
Comment (praseodym) [reply frontend/app/module/data_entry/page/FinaliseElectionPage.tsx:61]: Agreed, let me see if it's easy to do. We should've picked this up when the icon colouring was changed though.

File: frontend/app/module/NotAvailableInMock.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  return (
+    <>
+      <NavBar>
+        <Link to={"/overview"}>Overzicht</Link>
+      </NavBar>
+
+      <main>
+        <article>
+          Deze pagina is helaas niet beschikbaar in deze demo-versie. Ga naar het{" "}
```
[the remark this answers is not in the document]
Comment (praseodym) [reply frontend/app/module/NotAvailableInMock.tsx:20]: Our d2d tests should cover this. I don't think we have one for the current finalisation page though, we should add one in #420.

File: frontend/app/module/data_entry/page/FinaliseElectionPage.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (praseodym) [reply frontend/app/module/data_entry/page/FinaliseElectionPage.tsx]: Done in e2ffe82f017b2c22f163e5284636582f000af011.

File: frontend/app/test/unit/setup.ts [hunk 6]
```diff
@@ -3,6 +3,7 @@ import { cleanup, configure } from "@testing-library/react";
 import { afterEach, beforeAll, vi } from "vitest";
 import failOnConsole from "vitest-fail-on-console";
 
+import { resetDatabase } from "../../../lib/api-mocks/Database.ts";
```
Comment (Lionqueen94) [inline frontend/app/test/unit/setup.ts:6]:
```suggestion
import { resetDatabase } from "@kiesraad/api-mocks";
```

Comment (praseodym) [reply frontend/app/test/unit/setup.ts:6]: Good suggestion, simplified in cab555b3fa6c6d77069396d2b8abdeed52cd5475.

**Review by Lionqueen94 — APPROVED** [review]

**Review by cikzh — APPROVED** [review]

**Pull request merged**